src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -643,6 +629,7 @@ private static void SubstituteConstraintTypes(ImmutableArray<TypeSymbolWithAnnot
                 if (considerCustomModifiers)
                 {
                     if (!type1.TypeSymbol.Equals(type2.TypeSymbol, TypeCompareKind.ConsiderEverything.AddIgnoreDynamic(ignoreDynamic).AddIgnoreTupleNames(ignoreTupleNames)) ||
+                        !type1.CustomModifiers.SequenceEqual(type2.CustomModifiers) ||
```

Comment:
I don't follow. The previous line does compare only type symbols, but that doesn't handle custom modifiers, and `TypeWithAnnotations` does not declare `Equals`.